File: project_euler/2/2.go
package main

import (
	"fmt"
)

func main() {
	var a uint64 = 1
	var b uint64 = 2
	var sum uint64 = a + b
	var tmp_b uint64

	for b < 4000000 {
		tmp_b = b
		b = a + tmp_b
		a = tmp_b

		sum += b
	}

	fmt.Println("sum is:", sum)
}

File: project_euler/2/2.c
#include <stdio.h>

int main(void)
{
    unsigned long int a = 1;
    unsigned long int b = 2;
    unsigned long int sum = a + b;
    unsigned long int tmp_b;

    while (b < 4000000) {
        tmp_b = b;
        b = a + tmp_b;
        a = tmp_b;
        sum += b;
    }

    printf("sum is: %lu\n", sum);

    return 0;
}
